src/python/pants/backend/terraform/target_types.py

```diff
@@ -51,5 +57,71 @@ class TerraformModuleTarget(Target):
     )
 
 
+class TerraformBackendConfigField(OptionalSingleSourceField):
+    alias = "backend_config"
+    help = "Configuration to be merged with what is in the configuration file's 'backend' block"
+
+    def empty(self) -> TerraformBackendConfigField:
+        """Clear the backend config to ensure that."""
+        return TerraformBackendConfigField(None, self.address)
+
+
+class TerraformVarFilesField(MultipleSourcesField):
+    alias = "var_files"
+    expected_file_extensions = (".tfvars",)
+    help = generate_multiple_sources_field_help_message(
+        "Example: `var_files=['common.tfvars', 'prod.tfvars']`"
+    )
+
+
+class TerraformExtraArgs(StringSequenceField):
+    alias = "extra_args"
+    help = help_text(
+        """
+        Extra arguments for `terraform apply`
+        Example: `extra_args=["-var" "'var0=hihello'"]`"
+        """
+    )
+
+
+class TerraformDeploymentTarget(Target):
+    alias = "terraform_deployment"
+    core_fields = (
+        *COMMON_TARGET_FIELDS,
+        TerraformDependenciesField,
+        TerraformModuleSourcesField,
```

Comment:
Simply pointing out that the Terraform module has an implementation of the `tailor` goal that uses the default value of this field to create `terraform_module` targets, not `terraform_deployment` ones.

This can lead to user frustration since we encourage users to rely on `tailor` and Pants will be suggesting the wrong target type in this case. Not sure if there is a good way to consistently differentiate between a module and a deployment though.

Reply:
I agree that the expected results for tailor aren't clear. I haven't seen much from Hashicorp about distinguishing between root modules and non-root modules, except that root modules are the ones you want to run `apply` from. Comparing with Python, where `if __name__ == "__main__:"` is a clear signifier of a runnable target.
I think it is generally safe to propose adding a `terraform_deployment` target to all modules and have people remove them if they don't want them. But also, people might want to be more intentional about what deployments they create. 
Comparing to Helm: it only creates `HelmChartTarget`s, not `HelmDeploymentTarget`.